pkg/loadbalancer/maps/pressure_metrics.go

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package maps
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	"github.com/cilium/cilium/pkg/metrics"
+)
+
+type pressureMetricsParams struct {
+	cell.In
+
+	JobGroup        job.Group
+	MetricsRegistry *metrics.Registry
+	DB              *statedb.DB
+	Frontends       statedb.Table[*loadbalancer.Frontend]
+	Backends        statedb.Table[*loadbalancer.Backend]
+	Config          loadbalancer.Config
+	LBMaps          LBMaps
+}
+
+type pressureMetrics struct {
+	pressureMetricsParams
+
+	bpfMaps *BPFLBMaps
+	gauges  map[string]*metrics.GaugeWithThreshold
+}
+
+func registerPressureMetricsReporter(p pressureMetricsParams) {
+	if !p.Config.EnableExperimentalLB || p.Config.LBPressureMetricsInterval == 0 {
+		return
+	}
+
+	bpfMaps, ok := p.LBMaps.(*BPFLBMaps)
+	if !ok {
+		return
+	}
+
+	pm := &pressureMetrics{
+		pressureMetricsParams: p,
+		gauges:                map[string]*metrics.GaugeWithThreshold{},
+		bpfMaps:               bpfMaps,
+	}
+
+	// Update the metrics only once every 5 minutes as we need to iterate over all
+	// keys in each map to extract the count. If we're testing update the metrics
+	// frequently.
+	updateInterval := p.Config.LBPressureMetricsInterval
+	p.JobGroup.Add(job.Timer(
+		"pressure-metrics-reporter",
+		pm.report,
+		updateInterval,
+	))
+}
+
+func (pm *pressureMetrics) getGauge(mapName string) *metrics.GaugeWithThreshold {
+	mapName = strings.TrimPrefix(mapName, metrics.Namespace+"_")
+	if g, found := pm.gauges[mapName]; found {
+		return g
+	}
+	// Create a new metric. This will be registered/unregistered on-demand based on
+	// the threshold. E.g. if map pressure is 0.0 it won't appear in metrics.
+	g := pm.MetricsRegistry.NewBPFMapPressureGauge(mapName, 0.0)
+	pm.gauges[mapName] = g
+	return g
+}
+
+func (pm *pressureMetrics) report(ctx context.Context) error {
+	var errs error
+
+	// openMaps is immutable after Start(), so safe to access here.
+	created, _ := pm.bpfMaps.allMaps()
+	for _, desc := range created {
+		m := *desc.target
+		name := m.Name()
+
+		if m.Type() != ebpf.Hash {
```

Comment:
Do we need pressure on any LPMTrie maps?

Reply:
Ah yeah I guess we could report the source range map pressure too. Good catch.

Reply:
```
time=2025-05-09T15:13:09Z level=error source=/go/src/github.com/cilium/cilium/vendor/github.com/cilium/hive/job/timer.go:203 msg="Timer job errored" module=agent.controlplane.loadbalancer-maps name=pressure-metrics-reporter func=maps.(*pressureMetrics).report error="count on cilium_lb4_source_range failed: failed to iterate map: map batch lookup: not supported\ncount on cilium_lb6_source_range failed: failed to iterate map: map batch lookup: not supported" (1 occurrences)
```

Ah this won't work for LPMTries. I'll revert and only check hash maps.